Lib/Inc/CFIFOBuffer.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file CFIFOBuffer.h
+ */
+
+/*
+ *  Created on: Mar 28, 2022
+ *      Author: niuslar
+ */
+
+#ifndef CFIFOBUFFER_H_
+#define CFIFOBUFFER_H_
+
+#include "main.h"
+
+template <typename T, uint8_t BUF_SIZE>
```

Comment:
uint8_t limits buffer size to 255 items. 
Suggestions : unit32_t or uint16_t

Reply:
Thanks, fixed